File: Documentation/3.0/Samples/java/src/test/java/net/ravendb/clientapi/session/howto/Refresh.java
package net.ravendb.clientapi.session.howto;

import static org.junit.Assert.assertEquals;
import net.ravendb.client.IDocumentSession;
import net.ravendb.client.IDocumentStore;
import net.ravendb.client.document.DocumentStore;
import net.ravendb.samples.northwind.Employee;


public class Refresh {
  @SuppressWarnings("unused")
  private interface IFoo {
    //region refresh_1
    public <T> void refresh(T entity);
    //endregion
  }

  public Refresh() throws Exception {
    try (IDocumentStore store = new DocumentStore()) {
      try (IDocumentSession session = store.openSession()) {
        //region refresh_2
        Employee employee = session.load(Employee.class, "employees/1");
        assertEquals("Doe", employee.getLastName());

        // LastName changed to 'Shmoe'

        session.advanced().refresh(employee);
        assertEquals("Shmoe", employee.getLastName());
        //endregion
      }
    }
  }
}

File: Documentation/5.4/Samples/nodejs/client-api/session/howTo/refresh.js
import * as assert from "assert";
import { DocumentStore } from "ravendb";

const store = new DocumentStore();
const session = store.openSession();

async function refresh() {
    {
        //region refresh_2
        const employee = await session.load("employees/1");
        assert.strictEqual("Doe", employee.lastName);

        // lastName changed to 'Shmoe'

        session.advanced.refresh(employee);

        assert.strictEqual("Shmoe", employee.lastName);
        //endregion
    }
}

{
    let entity;
    //region refresh_1
    session.advanced.refresh(entity);
    //endregion
}
